src/visualizers/SeqColor.ts

```diff
@@ -0,0 +1,344 @@
+    name = 'Sequence Color'
+    n = 40
+
+    params = {
+        n: {
+            value: this.n,
+            forceType: 'integer',
+            displayName: 'number of terms',
+            required: true,
+        },
+    }
+
+    private currentNum = 0
+    private X = 0
+    private Y = 0
+    private boxSizeX = 800
+    private boxSizeY = 90
+    //private boxSize = this.sketch.createVector(800, 90)
+    private canvasSizeX = 800
+    private canvasSizeY = 800
+    private subG = this.sketch.createGraphics(this.boxSizeX, this.boxSizeY)
+    private subL = this.sketch.createGraphics(
+        this.canvasSizeX,
+        this.canvasSizeY
+    )
+    private boxIsShow = false
+    private primeNum: number[] = []
+    private countPrime = 0
+    private firstDraw = true
+    private showLabel = false
+    private brightAdjust = 100
+
+    initialize(sketch: p5, seq: SequenceInterface) {
+        super.initialize(sketch, seq)
+        this.sketch = sketch
+        this.seq = seq
+        this.currentNum = seq.first
+        this.X = 0
+        this.Y = 0
+        this.ready = true
+    }
+
+    setup() {
+        this.sketch.background('black')
+        this.sketch.colorMode(this.sketch.HSB, 360, 100, 100)
+        this.sketch.frameRate(30)
+
+        this.firstDraw = true
+
+        // Set position of the circle
+        this.X = this.canvasSizeX / this.n + 30
+        this.Y = this.canvasSizeX / this.n + 50
+
+        // Obtain all prime numbers from the sequence
+        for (let i = this.seq.first; i < this.n; i++) {
+            const checkCurrentPrime = Number(this.seq.getElement(i))
+            if (
+                this.isPrime(checkCurrentPrime)
+                && !this.primeNum.includes(checkCurrentPrime)
+            ) {
+                this.primeNum.push(checkCurrentPrime)
+                this.countPrime += 1
+            }
+        }
+    }
+
+    draw() {
+        if (this.firstDraw == true && this.currentNum < this.n) {
+            const currentElement = Number(
+                this.seq.getElement(this.currentNum++)
+            )
+            this.drawCircle(currentElement)
+
+            this.changePosition()
+
+            // Check if drawing finished
+            if (this.currentNum >= this.n) {
+                this.firstDraw = false
+            }
+            this.sketch.noStroke()
+        } else {
+            // Monitor keyboard events after finishing drawing
+            // Bug: multiple clicks detected when only one click happened
+            this.keyboardEvents()
+        }
+    }
+
+    keyboardEvents() {
+        // Show description box when the left arrow key is pressed
+        if (this.sketch.keyIsDown(this.sketch.LEFT_ARROW)) {
+            if (this.boxIsShow == true && this.subG != null) {
+                this.boxIsShow = false
+                this.undrawBox()
+            } else {
+                this.boxIsShow = true
+                this.drawBox()
+            }
+        }
+
```

Comment:
I've noticed what I think is a bug. Try the visualizer with the sequence from formula n, n+1, n+2, n+3 etc. The circles should just be "shifting left" between each of these. But the first circle is not fuzzy, it looks like a single disk, e.g. n+2. So something funny is happening with the first circle. (I've added this comment near where the disc is calculated, but I don't know what the bug is, I may be in the wrong place in the code.)

Reply:
This took a lot to track down!  It was because we needed to call `this.sketch.noStroke()` -- it was drawing the circle but with a black border that overwrote the colour of the last, larger circle each time.  So it was there, but very hard to see.  Somewhere else `noStroke()` gets set before it draws the next circle.  This is fixed in the most recent commit.  Please check it out.

Reply:
OK, going from n+2 to n+3 doesn't just shift left, it also changes colors because the primes are colored in the order they occur in the sequence, not in numerical order. But I think that is intentional, and I think it is working. Resolving.